File: panaderia-backend/src/main/java/com/besoft/panaderia/service/MaestraService.java
package com.besoft.panaderia.service;

import java.util.List;

import com.besoft.panaderia.dto.request.MaestraBuscarRequest;
import com.besoft.panaderia.dto.request.MaestraRequest;
import com.besoft.panaderia.dto.response.FileResponse;
import com.besoft.panaderia.dto.response.MaestraResponse;
import com.besoft.panaderia.dto.response.OutResponse;

public interface MaestraService {

	public OutResponse<List<MaestraResponse>> listarMaestra(MaestraBuscarRequest req);

	public OutResponse<MaestraResponse> registrarMaestra(MaestraRequest req);

	public OutResponse<MaestraResponse> modificarMaestra(MaestraRequest req);

	public OutResponse<MaestraResponse> eliminarMaestra(MaestraRequest req);
	
	public OutResponse<FileResponse> reporteXlsxListarMaestra(MaestraBuscarRequest req);
}

File: panaderia-frontend/src/main/panaderia-frontend/src/app/modules/intranet/services/maestra.service.ts
import { Inject, Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { MaestraRequest } from '../dto/request/maestra.request';
import { Observable } from 'rxjs';
import { OutResponse } from '../dto/response/out.response';
import { MaestraResponse } from '../dto/response/maestra.response';
import { environment } from 'src/environments/environment';
import { MaestraBuscarRequest } from '../dto/request/maestra-buscar.request';
import { UsuarioService } from 'src/app/core/services/usuario.service';

@Injectable()
export class MaestraService {

  constructor(private http: HttpClient,
    @Inject(UsuarioService) private user: UsuarioService) { }

  public listarMaestra(request: MaestraBuscarRequest): Observable<OutResponse<MaestraResponse[]>> {
    let httpOptions = {
      headers: new HttpHeaders({
        'Content-type': 'application/json; charset=utf-8',
        authorization: 'Bearer ' + this.user.access_token
      })
    };
    return this.http.post<OutResponse<MaestraResponse[]>>(`${environment.WsPanaderiaBackend}/maestra/listarMaestra`, request, httpOptions);
  }

  public registrarMaestra(request: MaestraRequest): Observable<OutResponse<MaestraResponse>> {
    let httpOptions = {
      headers: new HttpHeaders({
        'Content-type': 'application/json; charset=utf-8',
        authorization: 'Bearer ' + this.user.access_token
      })
    };
    return this.http.post<OutResponse<MaestraResponse>>(`${environment.WsPanaderiaBackend}/maestra/registrarMaestra`, request, httpOptions);
  }

  public modificarMaestra(request: MaestraRequest): Observable<OutResponse<MaestraResponse>> {
    let httpOptions = {
      headers: new HttpHeaders({
        'Content-type': 'application/json; charset=utf-8',
        authorization: 'Bearer ' + this.user.access_token
      })
    };
    return this.http.post<OutResponse<MaestraResponse>>(`${environment.WsPanaderiaBackend}/maestra/modificarMaestra`, request, httpOptions);
  }

  public eliminarMaestra(request: MaestraRequest): Observable<OutResponse<MaestraResponse>> {
    let httpOptions = {
      headers: new HttpHeaders({
        'Content-type': 'application/json; charset=utf-8',
        authorization: 'Bearer ' + this.user.access_token
      })
    };
    return this.http.post<OutResponse<MaestraResponse>>(`${environment.WsPanaderiaBackend}/maestra/eliminarMaestra`, request, httpOptions);
  }
}
